### PR TITLE
Create cross platform Home dir and config file

### PR DESCRIPTION
**Pull request by nunojusto**
## Problem

To store several config variables we need a config file

## Solution

Creation of a cross-platform home dir and a way to load variables from a JSON file

## Checklist

- [x] I have ran the tests locally.
- [x] I branched off of the `develop` branch and not `master`.
- [x] I did not change the `VERSION` file.

### CODE REVIEW

File: cli/config/config.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/atrox/homedir"
+	"github.com/spf13/viper"
+)
+
+const (
+	confDir = "~/.neo-local"
```
Comment (revett) [inline cli/config/config.go]: Best practice is to use full words in variable names.

File: cli/config/config.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	viper.SetConfigFile(filename)
+
+	log.Printf(" %s", filename)
+
+	err := viper.ReadInConfig()
+	if err != nil {
+		viper.Set("initialSetup", true)
+		viper.SafeWriteConfigAs(filename)
+	}
+	return nil
```
Comment (revett) [inline cli/config/config.go]:
```suggestion

	return nil
```

File: cli/config/config.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// DirPath retrieves the config folder complete path
+func DirPath() (string, error) {
+	d, err := homedir.Expand(confDir)
+	if err != nil {
+		return "", err
+	}
```
Comment (revett) [inline cli/config/config.go]:
```suggestion

	}
```

File: cli/config/config.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	if _, err := os.Stat(d); os.IsNotExist(err) {
+		err = os.MkdirAll(d, 0755)
+		if err != nil {
+			return "", err
+		}
+	}
+	return d, nil
```
Comment (revett) [inline cli/config/config.go]:
```suggestion

	return d, nil
```

File: cli/config/config.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	if err != nil {
+		viper.Set("initialSetup", true)
+		viper.SafeWriteConfigAs(filename)
+	}
+	return nil
+}
+
+// DirPath retrieves the config folder complete path
+func DirPath() (string, error) {
+	d, err := homedir.Expand(confDir)
```
Comment (revett) [inline cli/config/config.go]: Use full word variable names.

File: cli/config/config.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+// File retrieves the config file complete path
+func File() (string, error) {
+	dirPath, err := DirPath()
+	if err != nil {
+		return "", err
+	}
+
+	confFilePath := fmt.Sprintf("%s/config.json", dirPath)
+
+	return confFilePath, nil
```
Comment (revett) [inline cli/config/config.go]:
```suggestion
	return fmt.Sprintf("%s/config.json", dirPath), nil
```

File: cli/config/config.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// File retrieves the config file complete path
+func File() (string, error) {
+	dirPath, err := DirPath()
+	if err != nil {
+		return "", err
+	}
+
+	confFilePath := fmt.Sprintf("%s/config.json", dirPath)
```
Comment (revett) [inline cli/config/config.go]:
```suggestion
```

File: cli/config/config.go (hunk 1 above)
Comment (nunojusto) [reply cli/config/config.go]: done

File: cli/config/config.go (hunk 2 above)
Comment (nunojusto) [reply cli/config/config.go]: done

File: cli/config/config.go (hunk 3 above)
Comment (nunojusto) [reply cli/config/config.go]: done

File: cli/config/config.go (hunk 4 above)
Comment (nunojusto) [reply cli/config/config.go]: done

File: cli/config/config.go (hunk 5 above)
Comment (nunojusto) [reply cli/config/config.go]: done

File: cli/config/config.go (hunk 6 above)
Comment (nunojusto) [reply cli/config/config.go]: yeah, makes total sense

File: cli/config/config.go (hunk 7 above)
Comment (nunojusto) [reply cli/config/config.go]: done

**Pull request merged**